### PR TITLE
Remove usage of werkzeug.urls.url_encode

### LINKED ISSUE #1910: Fix DeprecationWarning from usage of werkzeug.urls.url_encode
From the latest, at time of writing, CI run on develop:
```
  /__w/MSS/MSS/mslib/msui/mscolab_version_history.py:140: DeprecationWarning: 'werkzeug.urls.url_encode' is deprecated and will be removed in Werkzeug 2.4. Use 'urllib.parse.urlencode' instead.
    query_string = url_encode({"named_version": named_version_only})
```

### PR DESCRIPTION
**Pull request by matrss**
Replaces it with `urllib.parse.urlencode`. Fixes #1910.

### CODE REVIEW

**Comment by matrss** [comment]
I don't get why this "AttributeError: module 'PyQt5.QtCore' has no attribute 'Slot'" is happening here, but not on develop. I did not change anything remotely related to that and `PyQt5.QtCore` simply does not have this attribute. See: https://stackoverflow.com/questions/50459672/attributeerror-module-pyqt5-qtcore-has-no-attribute-slot

Both CI runs report the same pyqt version though...

**Comment by matrss** [comment]
OK, so it looks like the `Slot` attribute must be coming from
```
lib/python3.10/site-packages/matplotlib/backends/qt_compat.py
94:        QtCore.Slot = QtCore.pyqtSlot
106:        QtCore.Slot = QtCore.pyqtSlot
```
in the CI run on develop. I am not sure why that is used in develop but not in this PR pipeline though.

Anyway, I think we should change all occurrences of `QtCore.Slot` to `QtCore.pyqtSlot`.

**Comment by matrss** [comment]
```
>>> werkzeug.urls.url_encode({"test": None})
''
>>> urllib.parse.urlencode({"test": None})
'test=None'
```

That's the difference causing our tests to fail.

File: mslib/msui/mscolab_version_history.py [hunk 1]
```diff
@@ -134,10 +133,9 @@ def load_all_changes(self):
                 "token": self.token,
                 "op_id": self.op_id
             }
-            named_version_only = None
```
Comment (ReimarBauer) [inline mslib/msui/mscolab_version_history.py:137]: this was one of these abuse of None for boolean. The minimum change would be here to set it to False.

The coresponding server code should also become fixed, e.g. file_manager.get_all_changes

**Review by ReimarBauer — CHANGES_REQUESTED** [review]
see comment. 

related to https://github.com/Open-MSS/MSS/issues/1614

**Comment by ReimarBauer** [comment]
> ```
> >>> werkzeug.urls.url_encode({"test": None})
> ''
> >>> urllib.parse.urlencode({"test": None})
> 'test=None'
> ```
> 
> That's the difference causing our tests to fail.

Good catch !

File: mslib/msui/mscolab_version_history.py (hunk 1 above)
Comment (matrss) [reply mslib/msui/mscolab_version_history.py:137]: Done

**Review by ReimarBauer — APPROVED** [review]
Thx

**Comment by ReimarBauer** [comment]
@joernu76 when this is merged we need to patch our server

**Pull request merged**